platform-tests/src/test/java/org/junit/platform/engine/support/hierarchical/NodeTreeWalkerIntegrationTests.java

```diff
@@ -154,4 +192,18 @@ void test() {
 			}
 		}
 	}
+
+	@ResourceLock("a")
+	static class TestCaseWithResourceWriteLockOnClass {
+		@Test
+		void test() {
+		}
+	}
+
+	@ResourceLock(value = "a", mode = ResourceAccessMode.READ)
+	static class TestCaseWithResourceReadLockOnClass {
+		@Test
+		void test() {
+		}
+	}
```

Comment:
I think we should we're missing a test where the class has `@ResourceLock(value = "a", mode = ResourceAccessMode.READ)` and the test method has `@ResourceLock("a")`